Describe behavior of `copy_grants` (#3474)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-configs/snowflake-configs.md

</TabItem>
</Tabs>

<VersionBlock firstVersion="1.2">

## Copying grants

</VersionBlock>

<VersionBlock firstVersion="1.2" lastVersion="1.5">

> When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. The default value is `false`.

Reviewer: ```suggestion
When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. The default value for the `copy_grants` config is set to `false`.
```

</VersionBlock>

<VersionBlock firstVersion="1.6">

> When the `copy_grants` config is set to `true`, dbt will add the `copy grants` <Term id="ddl" /> qualifier when rebuilding tables and <Term id="view">views</Term>. After materialization, it will also apply the `grants` that are [configured](/reference/resource-configs/grants#configuring-grants). No grants are revoked when `copy_grants` config is set to `true`.

Reviewer: ```suggestion
When you set the `copy_grants` config to `true`, dbt adds the `copy grants` <Term id="ddl" /> qualifier when it rebuilds tables and <Term id="view">views</Term>. After materialization, it also applies the `grants` that are [configured](/reference/resource-configs/grants#configuring-grants). When the `copy_grants `config is set to `true`, no grants are revoked.
```

> When the `copy_grants` config is set to `false`, dbt will revoke any existing grants prior to applying the configured `grants`.

Reviewer: ```suggestion
When you set the `copy_grants` config to `false`, dbt revokes any existing grants before applying the configured `grants`.
```

> The default value is `false`.

Reviewer: ```suggestion
The default value for the `copy_grants` config is set to `false`.
```
